beacon-chain/sync/service.go

```diff
@@ -0,0 +1,134 @@
+package sync
+
+import (
+	"context"
+	"hash"
+
+	"github.com/prysmaticlabs/prysm/beacon-chain/types"
+	"github.com/sirupsen/logrus"
+)
+
+var log = logrus.WithField("prefix", "sync")
+
+// Service is the gateway and the bridge between the p2p network and the local beacon chain.
+// In broad terms, a new block is synced in 4 steps:
+//     1. Receive a block hash from a peer
+//     2. Request the block for the hash from the network
+//     3. Receive the block
+//     4. Send the block to the local chain to validate
```

Comment:
send the block to the local chain to validate sounds weird, how about something like `node verifies the block using block processing and state transition logic locally`

Reply:
hmm this is trying to state that the sync service will forward the block to the beacon chain, but the sync service itself doesn't do any real validation (for example sig. verification)

maybe `forward block to the beacon service for full validation`

Reply:
yep I agree with that change